README.rst
Introduction
------------
**EZ Checksum** is a graphical application for calculating *fingerprints* for files to allow easy verifification of the file's authenticity and integrity.

Examples
--------
Typical uses include:

* Verifying downloaded files.
* Verifying files on removable media.
* Checking for duplicate files.

Interface
---------
The interface is designed to be as simple as possible, providing ameans to generate a checksum for a single file, and to verify the checksum against a known value.

Common usability features such as drag and drop, tool tips and keyboard shortcuts are provided, along with a comprehensive user manual.

User Manual
-----------
The user manual is included in the `Help <../../../help/index.html>`_ directory.

Docs
----
Technical / developer documentation is written in reStructuredText (rst) format, with HTML documentation generated by `Sphinx <http://www.sphinx-doc.org>`__.

Code documentation uses `Numpy style doctrings <https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt>`__.

`Example Numpy Style Python Docstrings <http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html>`__.


CHANGELOG.rst
This document lists significant changes and bug fixes in EZ Checksum Lite.

0.2.2 - Jan 2018
-----------------
* Github repository created '<https://github.com/SteveDaulton/EZ-Checksum>'_.
* :doc:`license` files added.

0.2.1 - Jan 2018
----------------
* Documentation update.

0.2.0 - Jan 2018
----------------
* Tabbed interface dropped in preference to three separate applications.
    * EZ Checksum Lite:
        This application. Focussed on simplicity. For testing single files.
    * EZ Checksum Pro:
        Support for multiple file / directory testing. Import / Export of checksum files. Verification strings from URL and more.
    * EZ Checksum Compare:
        For comparing files / directories.

0.1.0 - Dec 2017
----------------
* Added changelog
* Added HTML technical documentation

0.0.3 - Dec 2017
----------------
* Added drag and drop files into 'Simple' tab.

0.0.1 - Nov 2017
----------------

* Added support for XXH32 and XXH64

0.0.0 - Oct 2017
----------------
* Core application created.

    * Support for MD5, SHA-1, SHA-224, SHA-256, SHA-384, SHA-512.
    * Supports single file, multiple file and directory selection.

calc.py
"""Main class for calculating checksums"""

import threading

from PyQt4 import QtCore

import hash_profiles as Hp
#import EZchecksum as EZ

class ChecksumThread(QtCore.QThread):
    """Worker thread to calculate checksums.

    Args:
        algorithm: Int. Index number of selected algorithm.
        data: A QListWidget or QLineEdit containing the file names to be processed.

    """
    def __init__(self, algorithm, data):
        QtCore.QThread.__init__(self)
        self.algorithm = algorithm
        self.data = data

        self.stop_flag = False # Set to True when we want to stop processing.
        self.file_list_item = None

    # Override the destructor:
    def __del__(self):
        self.wait()

    def get_hash(self, fname):
        """Calculate the checksum."""
        profile = Hp.HASH_TYPES[self.algorithm]
        # Create a new copy of the hasher so that it will go out of
        # scope and be deleted when the function ends.
        hasher = profile['hasher'].copy()
        info = QtCore.QFileInfo(fname)
        size = info.size()
        # Test for zero byte file
        if size == 0:
            self.emit(QtCore.SIGNAL('checksum(QString, QString)'),
                      fname, 'Error: Empty file.')
            return  # just bail

        # Process in blocks of at least 64k
        blocksize = max(65536, size / 100)
        progress_step = min(1.0, blocksize / float(size)) * 100
        progress = 0.0
        step = max(1.0, progress_step)
        percent = int(step)

        # Connect a SIGNAL from this loop to the progressBar.
        try:
            with open(fname, 'rb') as file_:
                buf = file_.read(blocksize)
                while (len(buf) > 0) and not self.stop_flag:
                    hasher.update(buf)
                    buf = file_.read(blocksize)
                    # Update progress bar when there's an integer increase in progress
                    progress += progress_step
                    if progress >= percent:
                        self.emit(QtCore.SIGNAL('updateProgressBar(int)'), min(100, percent))
                        percent = int(progress + step)
                if not self.stop_flag:
                    self.emit(QtCore.SIGNAL('checksum(QString, QString)'),
                              fname, str(hasher.hexdigest()))
                else:
                    self.emit(QtCore.SIGNAL('updateProgressBar(int)'), 0)
 
        except (IOError, ValueError):
            dialogs.warning(self, 'An I/O error or a ValueError occurred')


    def run(self):
        """Override of QThread run."""
        self.get_hash(self.data.text())


    def stop(self):
        """Stop thread gracefully."""
        self.stop_flag = True



prefs.py
"""Helper functions for reading and writing user preferences."""

from PyQt4 import QtCore

import hash_profiles as Hp


def read_settings(self):
    """Restore from last used (or default) settings"""
    # TODO: Validate settings
    self.settings = QtCore.QSettings('AudioNyq', 'EZ Checksum')

    # Restore GUI size and position
    geometry = self.setGeometry(100, 100, -1, -1)
    geometry = self.settings.value('Geometry', geometry).toByteArray()
    self.restoreGeometry(geometry)

    # The 'algorithm' parameter carries the HASH_TYPE index number.
    # This makes it easier to retrieve the profile as: Hp.HASH_TYPES[self.algorithm]
    # Default hash encoding is Sha 256
    algorithm = self.settings.value('Algorithm', 'SHA256').toString()
    try:
        algorithm_index = Hp.HASH_CODES[str(algorithm)]
    except KeyError:
        algorithm_index = Hp.HASH_CODES['SHA256']
    self.algorithm = algorithm_index
    self.hashChoiceButton.setCurrentIndex(self.algorithm)


    # Default directory for opening files
    self.default_open_dir = self.settings.value(
        'OpenDirectory', self.default_open_dir).toString()

    # Default directory for saving results
    self.default_save_dir = self.settings.value(
        'SaveDirectory', self.default_save_dir).toString()


def write_settings(self):
    """Write last used settings as human readable strings"""
    geometry = self.saveGeometry()
    self.settings.setValue('Geometry', geometry)

    # self.algorithm is the HASH_TYPE index number.
    self.settings.setValue('Algorithm', Hp.HASH_STRINGS[self.algorithm])

    self.settings.setValue('OpenDirectory', self.default_open_dir)
    self.settings.setValue('SaveDirectory', self.default_save_dir)
    self.settings.sync()


dialogs.py
"""Additional dialogs used by the application"""

from PyQt4.QtGui import QMessageBox as QMsgBox
from PyQt4.QtGui import QFileDialog
from PyQt4.QtGui import QDialog

def warning(parent, message):
    """A simple warning message, Returns True on OK."""
    return QMsgBox.warning(parent, "Warning", message) == QMsgBox.Ok

def about(parent, version):
    """About dialog"""
    message = (r"""<strong>EZ Checksum version {version}</strong><br>
&nbsp;&nbsp;&nbsp;December 2017.<br>
&nbsp;&nbsp;&nbsp;<i>by Steve Daulton.</i>
<p>Website: <a href="https://easyspacepro.com">EasySpacePro.com</a><br>
Released under terms of the
<a href="https://www.gnu.org/licenses/gpl.html">GPLv3</a>.</p>""")
    message = message.format(version=version)
    QMsgBox.about(parent, "EZ Checksum", message)


def critical(parent, message):
    """A simple error message, Returns True on OK."""
    return QMsgBox.critical(parent, "Error", message) == QMsgBox.Ok

def dialog(message, title='Warning', more='', details='', choice=False):
    """A simple message box.
    Returns True if Ok / Yes, otherwise False.
    """
    msg_box = QMsgBox()
    msg_box.setIcon(QMsgBox.Warning)
    msg_box.setText(message)
    msg_box.setWindowTitle(title)
    if more:
        msg_box.setInformativeText(more)
    if details:
        msg_box.setDetailedText(details)
    msg_box.setEscapeButton(QMsgBox.Cancel)
    if choice:
        msg_box.setStandardButtons(QMsgBox.Yes | QMsgBox.No)
        return msg_box.exec_() == QMsgBox.Yes
    else:
        msg_box.setStandardButtons(QMsgBox.Ok)
        return msg_box.exec_() == QMsgBox.Ok

def modeless_dialog(parent, message, title='Warning', more='', details=''):
    """A simple non-modal message box"""
    msg_box = QMsgBox(parent)
    msg_box.setIcon(QMsgBox.Warning)
    msg_box.setWindowTitle(title)
    msg_box.setText(message)
    if more:
        msg_box.setInformativeText(more)
    if details:
        msg_box.setDetailedText(details)
    msg_box.setStandardButtons(QMsgBox.Ok)
    # In modeless QMessageBox, setEscapeButton must be set after setStandardButtons
    msg_box.setEscapeButton(QMsgBox.Ok)
    msg_box.show()

def save_results(parent, text):
    """Save results to file dialog"""
    dlog = QFileDialog()
    dlog.setAcceptMode(QFileDialog.AcceptSave)
    dlog.setFilter('Text Files (*.txt);;Log Files (*.log);;All Files (*)')
    dlog.setFileMode(QFileDialog.AnyFile)
    dlog.setDefaultSuffix('txt')
    dlog.setDirectory(parent.default_save_dir)
    # Check user didn't cancel
    if dlog.exec_() == QDialog.Accepted:
        parent.default_save_dir = dlog.directory().path()
        # Ensure '.log' suffix if Log File filter selected and no filter provided.
        # On Windows / Mac perhaps we should force the file extension?
        if dlog.selectedNameFilter() == 'Log Files (*.log)':
            dlog.setDefaultSuffix('log')
        try:
            _file = open(dlog.selectedFiles()[0], 'w')
        except IOError as err:
            detail_msg = 'I/O error: {0}\n{1}'.format(err.errno, err.strerror)
            dialog('File write error.', title='Error', details=detail_msg)
            return
        _file.write(text)
        _file.close()


validate.py
#!/usr/bin/env python

"""Validate module contains functions handling validation / verification."""

import sys
from threading import Thread
import time
#import re

#import dialogs
import hash_profiles as Hp


def hash_from_line(line):
    """Extract hash string from line of text.

    Return
    ------
        tuple or None
            Tuple in the form: (algorithm index, hash)
    """
    index = 0
    for profile in Hp.HASH_TYPES:
        match = profile['regex'].search(line)
        if match:
            return (index, match.group())
        index += 1
    return None


def set_validator(parent, text):
    """Set and disable the hashChoiceButton.

        Does a very fast length check first
  
        Return
        ------
            True if valid verifcation string, else False.
    """
    if len(text) in Hp.HASH_LENGTHS:
        hash_tuple = hash_from_line(text)
        if hash_tuple:
            parent.hashChoiceButton.setCurrentIndex(hash_tuple[0])
            parent.hashChoiceButton.setEnabled(False)
            #parent.set_hash_algorithm(hash_tuple[0])
            # Ensure that we have a clean hex string
            parent.validateLineEdit.setText(hash_tuple[1])
            parent.has_validator = True
            return
    else:
        parent.hashChoiceButton.setEnabled(True)
        parent.has_validator = False





hash_profiles.py
"""Provides dictionaries defining the available hashing profiles.

    Supported algorithms:

    - MD5
    - SHA-1
    - SHA-224
    - SHA-256
    - SHA-384
    - SHA-512
    - XXH32
    - XXH64

"""

import hashlib
import re
import xxhash

MD5 = {
    'name': 'MD5',
    'hasher': hashlib.md5(),
    'regex': re.compile(r'\b[a-f0-9]{32,32}\b', re.I)
    }
SHA1 = {
    'name': 'SHA-1',
    'hasher': hashlib.sha1(),
    'regex': re.compile(r'\b[a-f0-9]{40,40}\b', re.I)
    }
SHA224 = {
    'name': 'SHA-224',
    'hasher': hashlib.sha224(),
    'regex': re.compile(r'\b[a-f0-9]{56.56}\b', re.I)
    }
SHA256 = {
    'name': 'SHA-256',
    'hasher': hashlib.sha256(),
    'regex': re.compile(r'\b[a-f0-9]{64,64}\b', re.I)
    }
SHA384 = {
    'name': 'SHA-384',
    'hasher': hashlib.sha384(),
    'regex': re.compile(r'\b[a-f0-9]{96,96}\b', re.I)
    }
SHA512 = {
    'name': 'SHA-512',
    'hasher': hashlib.sha512(),
    'regex': re.compile(r'\b[a-f0-9]{128,128}\b', re.I)
    }
XXH32 = {
    'name': 'XXH32',
    'hasher': xxhash.xxh32(),
    'regex': re.compile(r'\b[a-f0-9]{8,8}\b', re.I)
    }
XXH64 = {
    'name': 'XXH64',
    'hasher': xxhash.xxh64(),
    'regex': re.compile(r'\b[a-f0-9]{16,16}\b', re.I)
    }

# A list of dicts that allow a specified hash profile to be selected by its index number.
HASH_TYPES = (MD5, SHA1, SHA224, SHA256, SHA384, SHA512, XXH32, XXH64,)
# The length of each hash type as hex string:
HASH_LENGTHS = (32, 40, 56, 64, 96, 128, 8,16,)
# A list of hashing type names for writing to saved settings.
HASH_STRINGS = ('MD5', 'SHA1', 'SHA224', 'SHA256', 'SHA384', 'SHA512', 'XXH32', 'XXH64',)
# HASH_CODES provide a reverse lookup of HASH_STRINGS
HASH_CODES = {type:HASH_STRINGS.index(type) for type in HASH_STRINGS}


gui.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'ez_gui.ui'
#
# Created by: PyQt4 UI code generator 4.11.4
#
# WARNING! All changes made in this file will be lost!

from PyQt4 import QtCore, QtGui

try:
    _fromUtf8 = QtCore.QString.fromUtf8
except AttributeError:
    def _fromUtf8(s):
        return s

try:
    _encoding = QtGui.QApplication.UnicodeUTF8
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig, _encoding)
except AttributeError:
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig)

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName(_fromUtf8("MainWindow"))
        MainWindow.resize(421, 545)
        self.centralwidget = QtGui.QWidget(MainWindow)
        self.centralwidget.setObjectName(_fromUtf8("centralwidget"))
        self.verticalLayout = QtGui.QVBoxLayout(self.centralwidget)
        self.verticalLayout.setObjectName(_fromUtf8("verticalLayout"))
        spacerItem = QtGui.QSpacerItem(400, 20, QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Minimum)
        self.verticalLayout.addItem(spacerItem)
        self.verticalLayout_1 = QtGui.QVBoxLayout()
        self.verticalLayout_1.setObjectName(_fromUtf8("verticalLayout_1"))
        self.horizontalLayout_1 = QtGui.QHBoxLayout()
        self.horizontalLayout_1.setObjectName(_fromUtf8("horizontalLayout_1"))
        self.label_1 = QtGui.QLabel(self.centralwidget)
        self.label_1.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignVCenter)
        self.label_1.setObjectName(_fromUtf8("label_1"))
        self.horizontalLayout_1.addWidget(self.label_1)
        self.label_2 = QtGui.QLabel(self.centralwidget)
        self.label_2.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignVCenter)
        self.label_2.setObjectName(_fromUtf8("label_2"))
        self.horizontalLayout_1.addWidget(self.label_2)
        spacerItem1 = QtGui.QSpacerItem(40, 20, QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        self.horizontalLayout_1.addItem(spacerItem1)
        self.fileAddButton = QtGui.QPushButton(self.centralwidget)
        self.fileAddButton.setObjectName(_fromUtf8("fileAddButton"))
        self.horizontalLayout_1.addWidget(self.fileAddButton)
        self.verticalLayout_1.addLayout(self.horizontalLayout_1)
        self.fileSelectLineEdit = QtGui.QLineEdit(self.centralwidget)
        self.fileSelectLineEdit.setToolTip(_fromUtf8("File may be dropped here"))
        self.fileSelectLineEdit.setObjectName(_fromUtf8("fileSelectLineEdit"))
        self.verticalLayout_1.addWidget(self.fileSelectLineEdit)
        self.verticalLayout.addLayout(self.verticalLayout_1)
        spacerItem2 = QtGui.QSpacerItem(400, 20, QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Minimum)
        self.verticalLayout.addItem(spacerItem2)
        self.verticalLayout_2 = QtGui.QVBoxLayout()
        self.verticalLayout_2.setObjectName(_fromUtf8("verticalLayout_2"))
        self.horizontalLayout_2 = QtGui.QHBoxLayout()
        self.horizontalLayout_2.setObjectName(_fromUtf8("horizontalLayout_2"))
        self.label_3 = QtGui.QLabel(self.centralwidget)
        self.label_3.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignVCenter)
        self.label_3.setObjectName(_fromUtf8("label_3"))
        self.horizontalLayout_2.addWidget(self.label_3)
        self.label_4 = QtGui.QLabel(self.centralwidget)
        self.label_4.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignVCenter)
        self.label_4.setWordWrap(False)
        self.label_4.setObjectName(_fromUtf8("label_4"))
        self.horizontalLayout_2.addWidget(self.label_4)
        spacerItem3 = QtGui.QSpacerItem(40, 20, QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        self.horizontalLayout_2.addItem(spacerItem3)
        self.verticalLayout_2.addLayout(self.horizontalLayout_2)
        self.validateLineEdit = QtGui.QLineEdit(self.centralwidget)
        self.validateLineEdit.setToolTip(_fromUtf8("Published checksum value to compare against.\n"
"(Right click or \'Ctrl + V\' to paste)"))
        self.validateLineEdit.setText(_fromUtf8(""))
        self.validateLineEdit.setObjectName(_fromUtf8("validateLineEdit"))
        self.verticalLayout_2.addWidget(self.validateLineEdit)
        self.verticalLayout.addLayout(self.verticalLayout_2)
        spacerItem4 = QtGui.QSpacerItem(400, 20, QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Minimum)
        self.verticalLayout.addItem(spacerItem4)
        self.horizontalLayout_3 = QtGui.QHBoxLayout()
        self.horizontalLayout_3.setObjectName(_fromUtf8("horizontalLayout_3"))
        self.label_5 = QtGui.QLabel(self.centralwidget)
        self.label_5.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignVCenter)
        self.label_5.setObjectName(_fromUtf8("label_5"))
        self.horizontalLayout_3.addWidget(self.label_5)
        self.label_6 = QtGui.QLabel(self.centralwidget)
        self.label_6.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignVCenter)
        self.label_6.setObjectName(_fromUtf8("label_6"))
        self.horizontalLayout_3.addWidget(self.label_6)
        spacerItem5 = QtGui.QSpacerItem(40, 20, QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        self.horizontalLayout_3.addItem(spacerItem5)
        self.goButton = QtGui.QPushButton(self.centralwidget)
        self.goButton.setEnabled(False)
        self.goButton.setObjectName(_fromUtf8("goButton"))
        self.horizontalLayout_3.addWidget(self.goButton)
        self.verticalLayout.addLayout(self.horizontalLayout_3)
        spacerItem6 = QtGui.QSpacerItem(400, 20, QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Minimum)
        self.verticalLayout.addItem(spacerItem6)
        self.verticalLayout_3 = QtGui.QVBoxLayout()
        self.verticalLayout_3.setObjectName(_fromUtf8("verticalLayout_3"))
        self.label_7 = QtGui.QLabel(self.centralwidget)
        self.label_7.setObjectName(_fromUtf8("label_7"))
        self.verticalLayout_3.addWidget(self.label_7)
        self.resultTextBrowser = QtGui.QTextBrowser(self.centralwidget)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.resultTextBrowser.sizePolicy().hasHeightForWidth())
        self.resultTextBrowser.setSizePolicy(sizePolicy)
        self.resultTextBrowser.setMinimumSize(QtCore.QSize(0, 0))
        self.resultTextBrowser.setTabStopWidth(80)
        self.resultTextBrowser.setObjectName(_fromUtf8("resultTextBrowser"))
        self.verticalLayout_3.addWidget(self.resultTextBrowser)
        self.progressBar = QtGui.QProgressBar(self.centralwidget)
        self.progressBar.setProperty("value", 0)
        self.progressBar.setObjectName(_fromUtf8("progressBar"))
        self.verticalLayout_3.addWidget(self.progressBar)
        self.horizontalLayout_4 = QtGui.QHBoxLayout()
        self.horizontalLayout_4.setObjectName(_fromUtf8("horizontalLayout_4"))
        spacerItem7 = QtGui.QSpacerItem(40, 20, QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        self.horizontalLayout_4.addItem(spacerItem7)
        self.hashChoiceButton = QtGui.QComboBox(self.centralwidget)
        self.hashChoiceButton.setToolTip(_fromUtf8("Hash algorithm is selected automatically\n"
"if optional \'Validation\' text set"))
        self.hashChoiceButton.setSizeAdjustPolicy(QtGui.QComboBox.AdjustToContentsOnFirstShow)
        self.hashChoiceButton.setObjectName(_fromUtf8("hashChoiceButton"))
        self.hashChoiceButton.addItem(_fromUtf8(""))
        self.hashChoiceButton.addItem(_fromUtf8(""))
        self.hashChoiceButton.addItem(_fromUtf8(""))
        self.hashChoiceButton.addItem(_fromUtf8(""))
        self.hashChoiceButton.addItem(_fromUtf8(""))
        self.hashChoiceButton.addItem(_fromUtf8(""))
        self.hashChoiceButton.addItem(_fromUtf8(""))
        self.hashChoiceButton.addItem(_fromUtf8(""))
        self.horizontalLayout_4.addWidget(self.hashChoiceButton)
        spacerItem8 = QtGui.QSpacerItem(40, 20, QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        self.horizontalLayout_4.addItem(spacerItem8)
        self.cancelButton = QtGui.QPushButton(self.centralwidget)
        self.cancelButton.setEnabled(False)
        self.cancelButton.setObjectName(_fromUtf8("cancelButton"))
        self.horizontalLayout_4.addWidget(self.cancelButton)
        self.resetButton = QtGui.QPushButton(self.centralwidget)
        self.resetButton.setEnabled(False)
        self.resetButton.setObjectName(_fromUtf8("resetButton"))
        self.horizontalLayout_4.addWidget(self.resetButton)
        self.verticalLayout_3.addLayout(self.horizontalLayout_4)
        self.verticalLayout.addLayout(self.verticalLayout_3)
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtGui.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 421, 27))
        self.menubar.setObjectName(_fromUtf8("menubar"))
        self.menuFile = QtGui.QMenu(self.menubar)
        self.menuFile.setObjectName(_fromUtf8("menuFile"))
        self.menuHelp = QtGui.QMenu(self.menubar)
        self.menuHelp.setObjectName(_fromUtf8("menuHelp"))
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtGui.QStatusBar(MainWindow)
        self.statusbar.setObjectName(_fromUtf8("statusbar"))
        MainWindow.setStatusBar(self.statusbar)
        self.actionSelect_File = QtGui.QAction(MainWindow)
        self.actionSelect_File.setObjectName(_fromUtf8("actionSelect_File"))
        self.actionSave_Result = QtGui.QAction(MainWindow)
        self.actionSave_Result.setObjectName(_fromUtf8("actionSave_Result"))
        self.actionUser_Manual = QtGui.QAction(MainWindow)
        self.actionUser_Manual.setObjectName(_fromUtf8("actionUser_Manual"))
        self.actionAbout = QtGui.QAction(MainWindow)
        self.actionAbout.setObjectName(_fromUtf8("actionAbout"))
        self.actionQuit = QtGui.QAction(MainWindow)
        self.actionQuit.setObjectName(_fromUtf8("actionQuit"))
        self.menuFile.addAction(self.actionSelect_File)
        self.menuFile.addAction(self.actionSave_Result)
        self.menuFile.addSeparator()
        self.menuFile.addAction(self.actionQuit)
        self.menuHelp.addAction(self.actionUser_Manual)
        self.menuHelp.addSeparator()
        self.menuHelp.addAction(self.actionAbout)
        self.menubar.addAction(self.menuFile.menuAction())
        self.menubar.addAction(self.menuHelp.menuAction())

        self.retranslateUi(MainWindow)
        self.hashChoiceButton.setCurrentIndex(3)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        MainWindow.setWindowTitle(_translate("MainWindow", "EZ Checksum", None))
        self.label_1.setText(_translate("MainWindow", "1)", None))
        self.label_2.setText(_translate("MainWindow", "Select the file to test.", None))
        self.fileAddButton.setToolTip(_translate("MainWindow", "Select file to test", None))
        self.fileAddButton.setText(_translate("MainWindow", "Select File", None))
        self.fileSelectLineEdit.setStatusTip(_translate("MainWindow", "No file selected", None))
        self.fileSelectLineEdit.setPlaceholderText(_translate("MainWindow", "File to test", None))
        self.label_3.setText(_translate("MainWindow", "2)", None))
        self.label_4.setText(_translate("MainWindow", "Enter the expected value.", None))
        self.validateLineEdit.setStatusTip(_translate("MainWindow", "No validation text entered", None))
        self.validateLineEdit.setPlaceholderText(_translate("MainWindow", "Validation (Optional)", None))
        self.label_5.setText(_translate("MainWindow", "3)", None))
        self.label_6.setText(_translate("MainWindow", "Click the \'Start\' button.", None))
        self.goButton.setToolTip(_translate("MainWindow", "Select file first", None))
        self.goButton.setStatusTip(_translate("MainWindow", "No file selected", None))
        self.goButton.setText(_translate("MainWindow", "Start", None))
        self.label_7.setText(_translate("MainWindow", "Test results below:", None))
        self.resultTextBrowser.setToolTip(_translate("MainWindow", "Output window", None))
        self.progressBar.setStatusTip(_translate("MainWindow", "Not running", None))
        self.hashChoiceButton.setStatusTip(_translate("MainWindow", "Current algorithm: SHA-256", None))
        self.hashChoiceButton.setItemText(0, _translate("MainWindow", "MD5", None))
        self.hashChoiceButton.setItemText(1, _translate("MainWindow", "SHA-1", None))
        self.hashChoiceButton.setItemText(2, _translate("MainWindow", "SHA-224", None))
        self.hashChoiceButton.setItemText(3, _translate("MainWindow", "SHA-256", None))
        self.hashChoiceButton.setItemText(4, _translate("MainWindow", "SHA-384", None))
        self.hashChoiceButton.setItemText(5, _translate("MainWindow", "SHA-512", None))
        self.hashChoiceButton.setItemText(6, _translate("MainWindow", "XXH32", None))
        self.hashChoiceButton.setItemText(7, _translate("MainWindow", "XXH64", None))
        self.cancelButton.setToolTip(_translate("MainWindow", "Stop processing", None))
        self.cancelButton.setText(_translate("MainWindow", "Cancel", None))
        self.resetButton.setToolTip(_translate("MainWindow", "Reset all", None))
        self.resetButton.setText(_translate("MainWindow", "Reset", None))
        self.menuFile.setTitle(_translate("MainWindow", "&File", None))
        self.menuHelp.setTitle(_translate("MainWindow", "&Help", None))
        self.actionSelect_File.setText(_translate("MainWindow", "Select File...", None))
        self.actionSave_Result.setText(_translate("MainWindow", "Save Result...", None))
        self.actionUser_Manual.setText(_translate("MainWindow", "Manual...", None))
        self.actionAbout.setText(_translate("MainWindow", "About...", None))
        self.actionQuit.setText(_translate("MainWindow", "Quit", None))


if __name__ == "__main__":
    import sys
    app = QtGui.QApplication(sys.argv)
    MainWindow = QtGui.QMainWindow()
    ui = Ui_MainWindow()
    ui.setupUi(MainWindow)
    MainWindow.show()
    sys.exit(app.exec_())

